src/widget/range/range-widget.js

```diff
@@ -84,7 +84,7 @@ class RangeWidget extends Widget {
         const html =
             `<div class="widget range-widget">
                 <div class="range-widget__wrap">
-                    <div class="range-widget__current"></div>
+                    <div class="range-widget__current">${this.props.value}</div>
```

Comment:
This is a totally fine solution.

It looks like the call to `this.update()` in `_init()` should have loaded the displayed current value as well. That way both the widget value as well as the current value box are set in the same way.

I think the bug appeared in the change listener when adding `if ( this.originalInputValue !== this.value )` to avoid unnecessary change events firing when setting `this.originalInputValue`.

Moving `this.current.textContent = this.value;` outside of that if statement would be an alternative fix to this issue. It avoids having to add a `prop` and sets the display value only in one location.

What do you think?

Reply:
I considered this approach as well. To be honest, I've spent so much time working with component libraries like React that I assumed `props` was the preferred solution for data passed into a component from outside context (albeit implicitly in this case). Which would you prefer?

Reply:
Thanks. Let's do the latter for the consistency with setting default values.